hare/messagevalidation.go

```diff
@@ -46,18 +47,39 @@ func (validator *MessageValidator) ValidateMessage(m *pb.HareMessage, k uint32)
 
 // verifies the message is contextually valid
 func isContextuallyValid(m *pb.HareMessage, k uint32) bool {
-	isSameIteration := iterationFromCounter(k) == iterationFromCounter(m.Message.K)
-	currentRound := k % 4
 	switch MessageType(m.Message.Type) {
 	case PreRound:
 		return true
+	case Notify:
+		return true
+	}
+
+	expIter := iterationFromCounter(k)
```

Comment:
replace all of this code (till the end of the switch) with a check whether k == m.Message.K